atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/CheckAndSetRequest.java

```diff
@@ -35,20 +40,33 @@
 
     public abstract Cell cell();
 
-    public abstract Optional<byte[]> oldValue();
+    @Nullable
+    @JsonProperty("oldValue")
+    protected abstract byte[] oldValueNullable();
```

Comment:
It looks like this is safe as far as serialization is concerned (https://github.com/FasterXML/jackson-modules-java8/tree/master/datatypes), so this should be fine.